csharp/downgrades/81a394bedda3d4747e36496c7be4d958424636d6/expressions.ql

```diff
@@ -0,0 +1,13 @@
+class Expression extends @expr {
+  string toString() { none() }
+}
+
+class TypeOrRef extends @type_or_ref {
+  string toString() { none() }
+}
+
+from Expression e, int k, int kind, TypeOrRef t
+where
+  expressions(e, k, t) and
+  if k = 135 then kind = 106 else kind = k
+select e, kind, t
```

Comment:
I imagine the below would require fewer write operations. I have no clue if that matters or not.
```
from Expression e, int k, int kind, TypeOrRef t
where
  expressions(e, k, t) and
  k = 135
select e, 106, t
```

Reply:
Maybe I have misunderstood the semantics of up- and downgrade scripts, but my impression is that `expressions.rel: run expressions.qlo` completely overwrites the expressions relation. If that is the case, will the above snippet not just drop everything from the expressions table that doesn't have kind = 135?

Reply:
I don't know how up/downgrade scripts are run. Let's not change anything here. Your script would work correctly in either case.

Reply:
From the description [here](https://github.com/github/codeql/blob/main/docs/prepare-db-upgrade.md), I think it uses *overwrite* semantics (a new `.rel`) file is created, meaning that we need to produce all tuples for the expressions relation.